Week8 - homework SQL/SQLQuery_2.sql

```diff
@@ -0,0 +1,157 @@
+CREATE DATABASE ZooManagement;
+
+USE ZooManagement;
+
+CREATE TABLE TicketTypes
+(
+Id INT IDENTITY(1,1) PRIMARY KEY,
+Subscription VARCHAR(20),
+Price INT,
+Validity VARCHAR(20),
+);
+
+CREATE TABLE Visitors
+(
+Id INT IDENTITY(1,1) PRIMARY KEY,
+LastName VARCHAR(50),
+FirstName VARCHAR(50),
+TicketId INT REFERENCES TicketTypes(Id),
+EntryDate Date,
+);
+
+CREATE TABLE Species
+(
+Id INT IDENTITY(1,1) PRIMARY KEY,
+Name VARCHAR(50),
+);
+
+CREATE TABLE Animals
+(
+Id INT IDENTITY(1,1) PRIMARY KEY,
+Name VARCHAR(50),
+Species INT REFERENCES Species(Id),
+);
+
+CREATE TABLE VisitorsFeedingAnimals
+(
+Id INT IDENTITY(1,1) PRIMARY KEY,
+VisitorId INT REFERENCES Visitors(Id),
+AnimalId INT REFERENCES Animals(Id),
+);
+
+CREATE TABLE Caretakers
+(
+Id INT IDENTITY(1,1) PRIMARY KEY,
+LastName VARCHAR(50),
+FirstName VARCHAR(50),
+SpeciesExpertise INT REFERENCES Species(Id),
```

Comment:
SpeciesId would make a better name for the foreign key even if "Species Expertise" sound better. In a database with tens / hundreds of tables is really hard to remember that SpeciesExpertise refer to a SpeciesId. It's kynf of a best practice to name the foreign keys as EntityId where Entity is the singular version (where a singular exists) to the tables this one is related to.